api/provider/aws/test_aws/service_create_test.go

```diff
@@ -0,0 +1,211 @@
+package test_aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestServiceCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "dpl_arn",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	defer provider.SetEntityIDGenerator("svc_id")()
+
+	loadBalancerInput := &elb.DescribeLoadBalancersInput{}
+	loadBalancerInput.SetLoadBalancerNames([]*string{aws.String("l0-test-lb_id")})
+	loadBalancerInput.SetPageSize(1)
+
+	listener := &elb.Listener{}
+	listener.SetInstancePort(80)
+
+	listenerDescription := &elb.ListenerDescription{}
+	listenerDescription.SetListener(listener)
+
+	listenerDescriptions := []*elb.ListenerDescription{
+		listenerDescription,
+	}
+
+	loadBalancerDescription := &elb.LoadBalancerDescription{}
+	loadBalancerDescription.SetListenerDescriptions(listenerDescriptions)
+	loadBalancerDescription.SetLoadBalancerName("l0-test-lb_id")
+
+	loadBalancerDescriptions := []*elb.LoadBalancerDescription{
+		loadBalancerDescription,
+	}
+
+	loadBalancerOutput := &elb.DescribeLoadBalancersOutput{}
+	loadBalancerOutput.SetLoadBalancerDescriptions(loadBalancerDescriptions)
+
+	mockAWS.ELB.EXPECT().
+		DescribeLoadBalancers(loadBalancerInput).
+		Return(loadBalancerOutput, nil)
+
+	taskDefinitionInput := &ecs.DescribeTaskDefinitionInput{}
+	taskDefinitionInput.SetTaskDefinition("dpl_arn")
+
+	portMapping := &ecs.PortMapping{}
+	portMapping.SetContainerPort(int64(80))
```

Comment:
Likewise here, you can use the new default `provider.DefaultLoadBalancerPort.ContainerPort` here:
`portMapping.SetContainerPort(int64(provider.DefaultLoadBalancerPort.ContainerPort))`

Reply:
Actually, disregard this. We're not talking about a LoadBalancer here, so that would be a little awkward to mix and match the default. Perhaps a new default could be defined?

Reply:
bivers++

Reply:
Is it worth it to create a default for a mocked return value of an AWS call in one test, though? I dunno, maybe it is.

Reply:
The default @jparsons04 actually can't exist, since it's a direct mapping of a specific task definition's container port to a specific loadbalancer's port.

Reply:
✅